kong/init.lua

```diff
@@ -1152,4 +1152,22 @@ function Kong.admin_header_filter()
 end
 
 
+function Kong.serve_status_api(options)
+  kong_global.set_phase(kong, PHASES.admin_api)
+
+  local options = options or {}
+
+  header["Access-Control-Allow-Origin"] = options.allow_origin or "*"
+
+  if ngx.req.get_method() == "OPTIONS" then
+    header["Access-Control-Allow-Methods"] = "GET, HEAD, PUT, PATCH, POST, DELETE"
+    header["Access-Control-Allow-Headers"] = "Content-Type"
+
+    return ngx.exit(204)
+  end
+
+  return lapis.serve("kong.status", options)
+end
```

Comment:
Now that the Admin API always returns latency headers, so should this new status API. Imho this would be the largest blocker for this PR today.